models/DataObject/ClassDefinition/Helper/Dao.php

```diff
@@ -119,8 +119,19 @@ protected function removeUnusedColumns(string $table, array $columnsToRemove, ar
             //if (!in_array($value, $protectedColumns)) {
             if (!in_array(strtolower($value), array_map('strtolower', $protectedColumns))) {
                 $dropColumns[] = 'DROP COLUMN `' . $value . '`';
+
+                if (str_contains($value, 'unit') === true) {
+                    $this->db->executeQuery(
+                        sprintf(
+                            'ALTER TABLE `%s` DROP FOREIGN KEY %s',
+                            $table,
+                            self::getForeignKeyName($table, $value)
+                        )
+                    );
+                }
```

Comment:
If we introduce a `foreignKeyExists` method, we could also use it here, right? 

And I am still not sure about the `str_contains` check. Could there be some side effects, if there are columns that just happen to have `unit` in their name?

Reply:
I've added the `foreignKeyExists` method and used it here.

I also changed it to use the Symfony String Component to not search for contains but ending with the added `__` I don't think one could have a column with `__unit` at the end.